### PR TITLE
Data app navigation

### LINKED ISSUE #25011: Implement basic app navigation

### PR DESCRIPTION
**Pull request by kulyk**
Resolves #25011, part of #24861

Implements basic data app navigation. When opening a data app from Metabase navigation sidebar, you'll now see a "Launch app" button at the top right. Once you click a button, the app is going to take over the whole Metabase navigation and the sidebar is going to now show a list of app pages (dashboards).

### To Verify

1. Follow steps from #24944 to create an app
2. Save/move a few dashboards into it (apps are collections _technically_)
3. Click the app name in Metabase nav sidebar
4. Ensure the URL is now `/apps/:id-:slug`
5. Click the "Launch app" button at the top right
6. Ensure you now see a list of dashboards in the sidebar instead of regular navigation bar
7. Ensure the URL is now `/a/:id-:slug`
8. Click on a dashboard in the app navigation bar
9. Ensure you can see the dashboard and the URL is now `/a/:appId/page/:dashboardId`
10. Click "Exit app", ensure you're back to /apps/:id-:slug` and the usual navigation sidebar is back

### Demo

https://user-images.githubusercontent.com/17258145/186692613-7c5fbce7-5da8-4725-98db-61473c9499da.mp4

### CODE REVIEW

File: frontend/src/metabase/collections/components/CollectionHeader/CollectionHeader.tsx [hunk 1]
```diff
@@ -32,6 +38,9 @@ const CollectionHeader = ({
         onUpdateCollection={onUpdateCollection}
       />
       <HeaderActions data-testid="collection-menu">
+        {isDataAppCollection(collection) && (
+          <LaunchDataAppButton collection={collection} />
+        )}
```
Comment (kulyk) [inline frontend/src/metabase/collections/components/CollectionHeader/CollectionHeader.tsx:43]: We're going to introduce a dedicated data app "landing" page, so this change is temporal

File: frontend/src/metabase/collections/components/CollectionHeader/LaunchDataAppButton.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  if (!dataApp) {
+    return null;
+  }
+  const path = Urls.dataApp(dataApp);
+  return (
+    <Button
+      icon="rocket"
+      onClick={() => onChangeLocation(path)}
+      small
+    >{t`Launch app`}</Button>
```
[the remark this answers is not in the document]
Comment (ranquild) [reply frontend/src/metabase/collections/components/CollectionHeader/LaunchDataAppButton.tsx:54]: `<Button as={Link} to={path} />`

File: frontend/src/metabase/nav/containers/MainNavbar/DataAppNavbarView.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      <DataAppActionsContainer>
+        <ButtonGroup>
+          <Tooltip tooltip={t`Add`}>
+            <DataAppActionButton icon="add" onlyIcon />
+          </Tooltip>
+          <Tooltip tooltip={t`Settings`}>
+            <DataAppActionButton icon="gear" onlyIcon />
+          </Tooltip>
+        </ButtonGroup>
+        <ExitDataAppButton
```
Comment (ranquild) [inline frontend/src/metabase/nav/containers/MainNavbar/DataAppNavbarView.tsx:91]: Maybe change it to a link, e.g. `as={Link} to={...}`

**Review by ranquild — APPROVED** [review]
A few things I've noticed
- I'm not redirected to the app after its creation and its confusing
- The empty state for an app is the same as collections, which confuses too
- After clicking `Launch app` the sidebar changes, but the button is still there and I initially though that it's a bug

Screenshot 2022-08-26 at 14 47 09
Screenshot 2022-08-26 at 14 48 25

File: frontend/src/metabase/collections/components/CollectionHeader/LaunchDataAppButton.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (kulyk) [reply frontend/src/metabase/collections/components/CollectionHeader/LaunchDataAppButton.tsx:54]: That's perfect, thank you!

**Comment by kulyk** [comment]
Thanks for your review @ranquild, all good calls 💪

**Comment by kulyk** [comment]
Addressed all the comments except that the "landing" page looks like a collection. Will replace it with a custom page a bit later

**Pull request merged**